sktime/data/drunkards_walk_simulator.py

```diff
@@ -0,0 +1,411 @@
+            The corresponding coordinate.
+        """
+        coord_j = np.floor(state / (self.grid_size[0]))
+        coord_i = state - coord_j * self.grid_size[0]
+        return coord_i, coord_j
+
+    def is_valid_coordinate(self, coord: Coordinate) -> bool:
+        r""" Validates if a coordinate is within bounds.
+
+        Parameters
+        ----------
+        coord : (i, j) tuple
+            The coordinate.
+
+        Returns
+        -------
+        is_valid : bool
+            Whether the coordinate is within bounds.
+        """
+        return (0 <= coord[0] < self.grid_size[0]) and (0 <= coord[1] < self.grid_size[1])
+
+    @property
+    def msm(self):
+        r""" Yields a :class:`MSM <sktime.markov.msm.MarkovStateModel>` which is parameterized with a transition matrix
+        corresponding to this setup.
+
+        Returns
+        -------
+        msm : sktime.markov.msm.MarkovStateModel
+            The corresponding Markov state model.
+        """
+        return self._msm
+
+    def walk(self, start: Coordinate, n_steps: int, stop: bool = True, return_states: bool = False, seed: int = -1):
+        r""" Simulates a random walk on the grid.
+
+        Parameters
+        ----------
+        start : (i, j) tuple
+            Start coordinate on the grid.
+        n_steps : int
+            Maximum number of steps to simulate.
+        stop : bool, default=False
+            Whether to stop the simulation once home or bar have been reached
+        return_states : bool, default=False
+            Whether to return states instead of coordinates. Default is False.
+        seed : int, default=-1
+            Random seed.
+
+        Returns
+        -------
+        random_walk : (n_steps, 2) ndarray
+            A random walk in coordinate space.
+        """
+        assert self.is_valid_coordinate(start), "Start must be within bounds."
+        if stop:
+            stopping_states = np.concatenate([self.home_state, self.bar_state])
+        else:
+            stopping_states = None
+        states = self.msm.simulate(n_steps, start=self.coordinate_to_state(start), stop=stopping_states, seed=seed)
+        if not return_states:
+            return np.array([self.state_to_coordinate(state) for state in states])
+        else:
+            return states
+
+    @staticmethod
+    @plotting_function
+    def plot_path(ax, path, intermediates: bool = True, color_lerp: bool = True, **plot_kw):
+        r""" Plots a path onto a drunkard's walk map. The path is interpolated with splines and, if desired, has a
+        linearly interpolated color along its path.
+
+        Parameters
+        ----------
+        ax : matplotlib axis
+            The axis to plot onto.
+        path : (N, 2) ndarray
+            The path in terms of 2-dimensional coordinates.
+        intermediates : bool, default=True
+            Whether to scatterplot the states that were visited in the path.
+        color_lerp : bool, default=True
```

Comment:
lerp?

Reply:
lerp is computer graphics jargon for linear interpolation